server/src/main/java/org/apache/uniffle/server/HealthCheck.java

```diff
@@ -78,23 +78,33 @@ public HealthCheck(AtomicBoolean isHealthy, ShuffleServerConf conf, List<Checker
     thread.setDaemon(true);
   }
 
-  @VisibleForTesting
-  void check() {
+
+  public void check() {
     for (Checker checker : checkers) {
       if (!checker.checkIsHealthy()) {
-        isHealthy.set(false);
+        shuffleServer.setServerStatus(ServerStatus.UNHEALTHY);
         ShuffleServerMetrics.gaugeIsHealthy.set(1);
         return;
       }
     }
     ShuffleServerMetrics.gaugeIsHealthy.set(0);
-    isHealthy.set(true);
+    if (shuffleServer.getServerStatus() == ServerStatus.DECOMMISSIONING) {
```

Comment:
Why do we need this?

Reply:
In ShuffleServer. After decommission is called, there are a large number of application is not complete case, ServerStatus state is DECOMMISSIONING, but we can't change the state of ServerStatus health thread detection

Reply:
why do we assign  `DECOMMISSIONING` to it if its origin value is `DECOMMISSIONING`? It seems that we don't change value.

Reply:
sorry, I made a mistake